frontend/packages/kubevirt-plugin/src/selectors/vm/selectors.ts

```diff
@@ -55,6 +53,52 @@ export const getVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[
 export const getDataVolumeTemplates = (vm: VMKind, defaultValue = []) =>
   _.get(vm, 'spec.dataVolumeTemplates') == null ? defaultValue : vm.spec.dataVolumeTemplates;
 
+export const getConfigMapVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[] =>
+  getVolumes(vm, defaultValue).filter((vol) => Object.keys(vol).includes('configMap'));
+
+export const getSecretVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[] =>
+  getVolumes(vm, defaultValue).filter((vol) => Object.keys(vol).includes('secret'));
+
+export const getServiceAccountVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[] =>
+  getVolumes(vm, defaultValue).filter((vol) => Object.keys(vol).includes('serviceAccount'));
+
+export const getEnvDiskVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[] => [
+  ...getConfigMapVolumes(vm, defaultValue),
+  ...getSecretVolumes(vm, defaultValue),
+  ...getServiceAccountVolumes(vm, defaultValue),
+];
+
+export const getConfigMapDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
+  getDisks(vm, defaultValue).filter(
+    (disk) => !!getConfigMapVolumes(vm).find((vol) => vol.name === disk.name),
+  );
+
+export const getSecretDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
+  getDisks(vm, defaultValue).filter(
+    (disk) => !!getSecretVolumes(vm).find((vol) => vol.name === disk.name),
+  );
+
+export const getServiceAccountDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
+  getDisks(vm, defaultValue).filter(
+    (disk) => !!getServiceAccountVolumes(vm).find((vol) => vol.name === disk.name),
+  );
+
+export const getEnvDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] => [
+  ...getConfigMapDisks(vm, defaultValue),
+  ...getSecretDisks(vm, defaultValue),
+  ...getServiceAccountDisks(vm, defaultValue),
+];
+
+export const getBootableDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
```

Comment:
Can't we just do a volume lookup and not calling getEnvDisks for each Disk every time?

Then we can just compare the VolumeType. No need for so many functions.